packages/frontend/core/src/modules/workbench/desktop-adapter.ts

```diff
@@ -0,0 +1,38 @@
+import { type Location } from 'history';
+import { useEffect } from 'react';
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { useLocation } from 'react-router-dom';
+
+import type { Workbench } from './workbench';
+
+export function useBindWorkbenchToDesktopRouter(
+  workbench: Workbench,
+  basename: string
+) {
+  const browserLocation = useLocation();
+
+  useEffect(() => {
+    const newLocation = browserLocationToViewLocation(
+      browserLocation,
+      basename
+    );
+    if (newLocation === null) {
+      return;
+    }
+
+    workbench.open(newLocation);
+  }, [basename, browserLocation, workbench]);
```

Comment:
I think `browserLocation` from `useLocation` will be a new reference on route change. May not use it in effect deps

Reply:
In desktop, we not really care about the browser history, we only listen it, and never modify it.

So only when the legacy navigation api is used (in most cases now), the browser route will change.

In the case of a browser route change, we simply push the new route into the active view.

Reply:
I am a bit concerned that putting route navigate in use effect may result in unwanted history items in the stack. Perhaps skipping route navigate in `workbench.open` if new route is exactly the same with the previous one?